### PR TITLE
catcli: init at 0.5.13

### PR DESCRIPTION
**Pull request by petersjt014**
###### Motivation for this change

It's a neat tool.

###### Things done

Added the package, included tests found and done automatically by the `buildPythonApplication` function. Works with both x86-64 and i686. Aarch64 does not build with pkgsCross for reasons that I'm not sure of yet, but it does build natively.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by petersjt014** [comment]
Had to undo/redo the commit to split adding the package and myself as a maintainer into separate commits.

File: pkgs/tools/filesystems/catcli/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  propagatedBuildInputs = [ docopt anytree ];
+
+  postPatch = '' patchShebangs . '';
+
+  meta = with stdenv.lib; {
+    description = "The command line catalog tool for your offline data";
+    homepage = "https://github.com/deadc0de6/catcli";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ petersjt014 ];
+    platforms = with platforms; [ x86_64-linux i686-linux ];
```
Comment (Evils-Devils) [inline pkgs/tools/filesystems/catcli/default.nix:28]: it builds on my pi...
```suggestion
    platforms = platforms.all;
```

File: pkgs/tools/filesystems/catcli/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "deadc0de6";
+    repo = pname;
+    rev = version;
+    sha256 = "04mrkqmm2c8fhi1h1hddc4yh3dqhcvkmcwzhj0ggn34v7wavgb5i";
+  };
+
+  propagatedBuildInputs = [ docopt anytree ];
+
+  postPatch = '' patchShebangs . '';
```
Comment (Evils-Devils) [inline pkgs/tools/filesystems/catcli/default.nix:21]: what necessitated this?
it seems to work without it (though i am not familiar with this tool)

File: pkgs/tools/filesystems/catcli/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+, fetchFromGitHub
+, buildPythonApplication
+, docopt, anytree 
+}:
+
+buildPythonApplication rec {
+
+  pname = "catcli";
+  version = "v0.5.13";
+
```
Comment (Evils-Devils) [inline pkgs/tools/filesystems/catcli/default.nix:11]: you've got some trailing whitespace here
```suggestion
, docopt, anytree
}:

buildPythonApplication rec {

  pname = "catcli";
  version = "v0.5.13";

```

**Review by evils — CHANGES_REQUESTED** [review]
none of this seems relevant to the grahamcofborg failure...

File: pkgs/tools/filesystems/catcli/default.nix (hunk 1 above)
Comment (petersjt014) [reply pkgs/tools/filesystems/catcli/default.nix:28]: Just tried and it builds natively on mine too.
Unfortuantely, the `raspberryPi` cross-build target still doesn't work (and I haven't tested it on anything else) so I'm hesitant to use `all` for now.

File: pkgs/tools/filesystems/catcli/default.nix (hunk 2 above)
Comment (petersjt014) [reply pkgs/tools/filesystems/catcli/default.nix:21]: [here's](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/patch-shebangs.sh#L14) where I was pointed to when I asked. Apparently it's recommended to catch subtle path-related bugs (which could come up later if new deps are introduced). I don't think it's doing much now but it's also not harming anything.

File: pkgs/tools/filesystems/catcli/default.nix (hunk 1 above)
Comment (petersjt014) [reply pkgs/tools/filesystems/catcli/default.nix:28]: For anyone wondering this is the build log for `raspberryPi`: http://0x0.st/iu-6.txt

File: pkgs/tools/filesystems/catcli/default.nix (hunk 2 above)
Comment (Evils-Devils) [reply pkgs/tools/filesystems/catcli/default.nix:21]: the `fixupPhase` already does this
going by the fact that without this line the wrapped script has a shebang to python in the nix store, i'm guessing it also runs for `buildPythonApplication`

**Comment by petersjt014** [comment]
The commit history's a tad ugly, but the package files are all tidily formatted and it works on all three listed platforms as advertised. It should be ready to merge.

**Review by evils — COMMENTED** [review]
besides the `postPatch` line bothering me for no good reason, this looks good to me
i'd love to know what grahamcofbork is on about...

**Review by cole-h — CHANGES_REQUESTED** [review]

**Review by Ma27 — CHANGES_REQUESTED** [review]
Please squash the last two commits together. Otherwise LGTM, just tested locally.

**Review by Ma27 — APPROVED** [review]

**Pull request merged**